contracts/mainnet/connectors/compound-iii/main.sol

```diff
@@ -0,0 +1,879 @@
+	}
+
+	/**
+	 * @dev Withdraw base/collateral asset or borrow base asset and transfer to 'to'.
+	 * @notice Withdraw base token or deposited token from Compound on behalf of an address and transfer to 'to'.
+	 * @param market The address of the market from where to withdraw.
+	 * @param token The address of the token to be withdrawn. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param to The address to which the borrowed assets are to be transferred.
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens withdrawn.
+	 */
+	function withdrawOnbehalf(
+		address market,
+		address token,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		(uint256 _amt, uint256 _setId) = _borrowOrWithdraw(
+			BorrowWithdrawParams({
+				market: market,
+				token: token,
+				from: address(0),
+				to: to,
+				amt: amt,
+				getId: getId,
+				setId: setId
+			})
+		);
+
+		_eventName = "LogWithdrawOnBehalf(address,address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(market, token, to, _amt, getId, _setId);
+	}
+
+	/**
+	 * @dev Withdraw base/collateral asset or borrow base asset from an account and transfer to 'to'.
+	 * @notice Withdraw base token or deposited token from Compound from an address and transfer to 'to'.
+	 * @param market The address of the market from where to withdraw.
+	 * @param token The address of the token to be withdrawn. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param from The address from where asset is to be withdrawed.
+	 * @param to The address to which the borrowed assets are to be transferred.
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens withdrawn.
+	 */
+	function withdrawFrom(
+		address market,
+		address token,
+		address from,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		(uint256 _amt, uint256 _setId) = _borrowOrWithdraw(
+			BorrowWithdrawParams({
+				market: market,
+				token: token,
+				from: from,
+				to: to,
+				amt: amt,
+				getId: getId,
+				setId: setId
+			})
+		);
+
+		_eventName = "LogWithdrawFrom(address,address,address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(market, token, from, to, _amt, getId, _setId);
+	}
+
+	/**
+	 * @dev Borrow base asset.
+	 * @notice Withdraw base token from Compound.
+	 * @param market The address of the market from where to withdraw.
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens withdrawn.
```

Comment:
@Richa-iitr While using borrow function, let's use borrow only instead of withdraw in comments.

Reply:
done.